modules/gltf/gltf_document.cpp

```diff
@@ -3136,6 +3136,64 @@ Error GLTFDocument::_parse_meshes(Ref<GLTFState> p_state) {
 				}
 			}
 
+			for (Ref<GLTFDocumentExtension> ext : document_extensions) {
+				for (const String &attrib_name : a.keys()) {
+					if (!attrib_name.begins_with("_")) {
+						continue;
+					}
+
+					const int custom_i = ext->remap_custom_attribute(attrib_name);
+					if (custom_i == -1) {
+						continue;
+					}
+
+					ERR_CONTINUE_MSG(custom_i < Mesh::ARRAY_CUSTOM0 || custom_i > Mesh::ARRAY_CUSTOM3, "Invalid MESH_CUSTOM index, ignoring attribute.");
+					ERR_CONTINUE_MSG(array[custom_i].get_type() != Variant::NIL, "Data already exists for MESH_CUSTOM index, ignoring attribute.");
+
+					bool format_okay = true;
+					Mesh::ArrayCustomFormat format{};
+					const Ref<GLTFAccessor> accessor = p_state->accessors[a[attrib_name]];
+					switch (accessor->accessor_type) {
+						case GLTFAccessor::TYPE_SCALAR: {
+							array[custom_i] = _decode_accessor_as_floats(p_state, a[attrib_name], true, indices_mapping);
+							format = Mesh::ARRAY_CUSTOM_R_FLOAT;
+						} break;
+						case GLTFAccessor::TYPE_VEC2: {
+							array[custom_i] = _decode_accessor_as_vec2(p_state, a[attrib_name], true, indices_mapping);
+							format = Mesh::ARRAY_CUSTOM_RG_FLOAT;
+						} break;
+						case GLTFAccessor::TYPE_VEC3: {
+							array[custom_i] = _decode_accessor_as_vec3(p_state, a[attrib_name], true, indices_mapping);
+							format = Mesh::ARRAY_CUSTOM_RGB_FLOAT;
+						} break;
+						case GLTFAccessor::TYPE_VEC4: {
+							array[custom_i] = _decode_accessor_as_floats(p_state, a[attrib_name], true, indices_vec4_mapping);
```

Comment:
Why are Vec3s decoded as Vector3, but Vec4s are not decoded as Vector4?

Reply:
There's not a decode_as_vec4 method, that's how it's done in the rest of the class. But come to think of it, they probably all need to be decoded as float, thanks for pointing that out.